librosa/display.py

```diff
@@ -1220,8 +1323,197 @@ def __coord_time(n, sr=22050, hop_length=512, **_kwargs):
+    samples-based view of the signal (`matplotlib.pyplot.step`) and an
+    amplitude-envelope view of the signal (`matplotlib.pyplot.fill_between`)
+    depending on the time extent of the plot's viewport.
+
+    More specifically, when the plot spans a time interval of less than `max_points /
+    sr` (by default, 1/2 second), the samples-based view is used, and otherwise a
+    downsampled amplitude envelope is used. 
+    This is done to limit the complexity of the visual elements to guarantee an
+    efficient, visually interpretable plot.
+
+    When using interactive rendering (e.g., in a Jupyter notebook or IPython
+    console), the plot will automatically update as the view-port is changed, either
+    through widget controls or programmatic updates.
+
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,) or (2,n)]
+        audio time series (mono or stereo)
+
+    sr : number > 0 [scalar]
+        sampling rate of ``y``
+
+    max_points : postive integer
+        Maximum number of samples to draw.  When the plot covers a time extent
+        smaller than `max_points / sr` (default: 1/2 second), samples are drawn.
+
+        If drawing raw samples would exceed `max_points`, then a downsampled
+        amplitude envelope extracted from non-overlapping windows of `y` is
+        visualized instead.  The parameters of the amplitude envelope are defined so
+        that the resulting plot cannot produce more than `max_points` frames.
+
+    x_axis : str or None
+        Display of the x-axis ticks and tick markers. Accepted values are:
+
+        - 'time' : markers are shown as milliseconds, seconds, minutes, or hours.
+                    Values are plotted in units of seconds.
+
+        - 's' : markers are shown as seconds.
+
+        - 'ms' : markers are shown as milliseconds.
+
+        - 'lag' : like time, but past the halfway point counts as negative values.
+
+        - 'lag_s' : same as lag, but in seconds.
+
+        - 'lag_ms' : same as lag, but in milliseconds.
+
+        - `None`, 'none', or 'off': ticks and tick markers are hidden.
+
+
+    ax : matplotlib.axes.Axes or None
+        Axes to plot on instead of the default `plt.gca()`.
+
+    offset : float
+        Horizontal offset (in seconds) to start the waveform plot
+
+    marker : string
+        Marker symbol to use for sample values. (default: no markers)
+
+    where : string, {'pre', 'mid', 'post'}
+        This setting determines how both waveform and envelope plots interpolate
+        between observations.
+
+        See `matplotlib.pyplot.step` for details.
+
+        Default: 'post'
+
+    label : string [optional]
+        The label string applied to this plot.
+        Note that the label
+
+    kwargs
+        Additional keyword arguments to `matplotlib.pyplot.fill_between` and
+        `matplotlib.pyplot.step`.
+
+        Note that only those arguments which are common to both functions will be
+        supported.
```

Comment:
i looked at the kwargs of step and fill_between and here are the kwargs in common (at least the ones most likely to be overriden by the user)
alpha, animated, color, label, linestyle, linewidth, snap

maybe worth documenting here for convenience?

Reply:
I'd rather not, in case it changes in the future and we get out of sync with mpl docs.